pkg/eval/builtin_fn_pred.go

```diff
@@ -305,6 +305,17 @@ func cmp(a, b any) ordering {
 				return more
 			}
 		}
+	case bool:
+		if b, ok := b.(bool); ok {
+			switch {
+			case a == b:
+				return equal
+			case !a: // `&& b` is implicit
```

Comment:
I find this to be a rare case where `a == false` is preferable to `!a`, as `a` is being tested for its value rather than being used as a condition.